src/term/transform/inline.rs

```diff
@@ -23,40 +23,44 @@ impl Book {
 
 impl Term {
   fn inline(&mut self, inlineables: &HashSet<Name>, defs: &IndexMap<Name, Definition>) {
-    match self {
-      Term::Ref { nam: def_name } => {
-        if inlineables.contains(def_name) {
-          *self = defs.get(def_name).unwrap().rules[0].body.clone();
+    let mut to_inline = vec![self];
+
+    while let Some(term) = to_inline.pop() {
+      match term {
+        Term::Ref { nam: def_name } => {
+          if inlineables.contains(def_name) {
+            *term = defs.get(def_name).unwrap().rules[0].body.clone();
```

Comment:
```suggestion
            *term = defs.get(def_name).unwrap().rule().body.clone();
```
This will also check that we only have 1 rule here